src/controllers/authenticationController.ts

```diff
@@ -88,7 +91,9 @@ export default class AuthenticationController {
   > {
     return withUserFromRequest(req, user =>
       withCatchAsInternalError(async () => {
-        const errorOrResponse = await SessionStorage.del(user.sessionToken);
+        const errorOrResponse = await SessionStorage.del(
+          user.sessions[0].token
```

Comment:
I don't understand this part, you should either:

* delete the current session
* delete all sessions

here instead you're deleting the first session returned by the query, not necessarily the current one (supposing I'm logged in from multiple browser sessions)